Shared/Samples/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningView.Model.swift

```diff
@@ -0,0 +1,154 @@
+        /// - Parameter map: The map which contains the tables from which the data source is constructed.
+        /// - Returns: Returns a configured IndoorsLocationDataSource created from the IPS position table.
+        private func createIndoorLocationDataSourceFromTables(map: Map) async throws -> IndoorsLocationDataSource? {
+            // Gets the positioning table from the map.
+            guard let positioningTable = map.tables.first(where: { $0.displayName == "IPS_Positioning" }) else { return nil }
+            // Creates and configures the query parameters.
+            let queryParameters = QueryParameters()
+            queryParameters.maxFeatures = 1
+            queryParameters.whereClause = "1 = 1"
+            // Queries positioning table to get the positioning ID.
+            let queryResult = try await positioningTable.queryFeatures(using: queryParameters)
+            guard let feature = queryResult.features().first(where: { _ in true }) else { return nil }
+            let serviceFeatureTable = positioningTable as! ServiceFeatureTable
+            let positioningID = feature.attributes[serviceFeatureTable.globalIDField] as? UUID
+            
+            // Gets the pathways layer (optional for creating the IndoorsLocationDataSource).
+            let pathwaysLayer = map.operationalLayers.first(where: { $0.name == "Pathways" }) as! FeatureLayer
+            // Gets the levels layer (optional for creating the IndoorsLocationDataSource).
+            let levelsLayer = map.operationalLayers.first(where: { $0.name == "Levels" }) as! FeatureLayer
+            
+            // Setting up IndoorsLocationDataSource with positioning, pathways tables and positioning ID.
+            return IndoorsLocationDataSource(
+                positioningTable: positioningTable,
+                pathwaysTable: pathwaysLayer.featureTable as? ArcGISFeatureTable,
+                levelsTable: levelsLayer.featureTable as? ArcGISFeatureTable,
+                positioningID: positioningID
+            )
+        }
+        
+        /// Updates the location when the indoors location datasource is triggered.
+        func updateDisplayOnLocationChange(locationUpdate: () -> Void) async throws {
+            for await location in locationDisplay.dataSource.locations {
+                // Since this listens for new location changes, it is important
+                // to ensure any blocking UI is dismissed once location updates begins.
+                // Floors in location are zero indexed however floorManager levels begin at one. Since
+                // it is necessary to display the same information to the user as the floor manager levelNumber
+                // one is added to the floor level value.
+                if let floorLevel = location.additionalSourceProperties[.floor] as? Int,
+                   (floorLevel + 1) != currentFloor {
```

Comment:
This is akin to a [Yoda condition](https://en.wikipedia.org/wiki/Yoda_conditions), which we avoid as a matter of style. Please use `floorLevel != currentFloor - 1` instead.